gcloud-java-storage/src/test/java/com/google/gcloud/storage/StorageImplTest.java

```diff
@@ -0,0 +1,830 @@
+    storage = StorageFactory.instance().get(optionsMock);
+    Capture<StorageRpc.BatchRequest> capturedBatchRequest = Capture.newInstance();
+    EasyMock.expect(storageRpcMock.batch(EasyMock.capture(capturedBatchRequest))).andReturn(res);
+    EasyMock.replay(storageRpcMock);
+    BatchResponse batchResponse = storage.apply(req);
+
+    // Verify captured StorageRpc.BatchRequest
+    List<Tuple<StorageObject, Map<StorageRpc.Option, ?>>> capturedToDelete =
+        capturedBatchRequest.getValue().toDelete;
+    List<Tuple<StorageObject, Map<StorageRpc.Option, ?>>> capturedToUpdate =
+        capturedBatchRequest.getValue().toUpdate;
+    List<Tuple<StorageObject, Map<StorageRpc.Option, ?>>> capturedToGet =
+        capturedBatchRequest.getValue().toGet;
+    for (int i = 0; i < capturedToDelete.size(); i++) {
+      assertEquals(toDelete.get(i), capturedToDelete.get(i).x());
+      assertEquals(deleteOptions.get(i), capturedToDelete.get(i).y());
+    }
+    for (int i = 0; i < capturedToUpdate.size(); i++) {
+      assertEquals(toUpdate.get(i), capturedToUpdate.get(i).x());
+      assertEquals(updateOptions.get(i), capturedToUpdate.get(i).y());
+    }
+    for (int i = 0; i < capturedToGet.size(); i++) {
+      assertEquals(toGet.get(i), capturedToGet.get(i).x());
+      assertEquals(getOptions.get(i), capturedToGet.get(i).y());
+    }
+
+    // Verify BatchResponse
+    for (BatchResponse.Result<Boolean> result : batchResponse.deletes()) {
+      assertTrue(result.get());
+    }
+    for (int i = 0; i < batchResponse.updates().size(); i++) {
+      assertEquals(toUpdate.get(i), batchResponse.updates().get(i).get().toPb());
+    }
+    for (int i = 0; i < batchResponse.gets().size(); i++) {
+      assertEquals(toGet.get(i), batchResponse.gets().get(i).get().toPb());
+    }
+  }
+
+  @Test
+  public void testReader() {
+    EasyMock.expect(optionsMock.storageRpc()).andReturn(storageRpcMock).times(2);
+    EasyMock.replay(optionsMock);
+    EasyMock.replay(storageRpcMock);
+    storage = StorageFactory.instance().get(optionsMock);
+    BlobReadChannel channel = storage.reader(BUCKET_NAME1, BLOB_NAME1);
+    assertNotNull(channel);
+    assertTrue(channel.isOpen());
+  }
+
+  @Test
+  public void testWriter() {
+    EasyMock.expect(optionsMock.storageRpc()).andReturn(storageRpcMock).times(2);
+    EasyMock.replay(optionsMock);
+    storage = StorageFactory.instance().get(optionsMock);
+    EasyMock.expect(storageRpcMock.open(BLOB_INFO1.toPb(), EMPTY_RPC_OPTIONS))
+        .andReturn("upload-id");
+    EasyMock.replay(storageRpcMock);
+    BlobWriteChannel channel = storage.writer(BLOB_INFO1);
+    assertNotNull(channel);
+    assertTrue(channel.isOpen());
+  }
+
+  @Test
+  public void testSignUrl() throws NoSuchAlgorithmException, InvalidKeyException,
+      SignatureException, UnsupportedEncodingException {
+    String account = "account";
+    KeyPair keyPair = KeyPairGenerator.getInstance("RSA").generateKeyPair();
```

Comment:
@aozarov It does not seem to be taking too long to me. By default it should generate a key of length 1024 which should take between 1 and 10 seconds on a relatively recent machine. If you prefer we can always embed private and public keys as strings constants.